MMOEngine/src/system/lang/String.cpp

```diff
@@ -494,7 +494,7 @@ String String::withCommas(int64 val) {
 	std::size_t i = 0;
 
 	for (auto pos = src.rbegin(); pos != src.rend(); ++pos) {
-		if (i % 3 == 0 && i != 0) {
+		if (i % 3 == 0 && i != 0 && !(val < 0 && i == src.length() - 1 && src[0] == '-')) {
```

Comment:
I would suggest checking if negative at the top, then use the absolute val and at the end add the sign '-' so you're not iterating all this extra logic for every single digit.

Reply:
Makes sense, my first attempt was just fixing it at the end.

Reply:
One more question.  I need a library for abs().   Should I include cmath, or Math.h?

Reply:
val = -val;

should work?

Reply:
Lol, I was over-complicating as usual.  Updated.